internal/app/vodtester/vodtester_app.go

```diff
@@ -0,0 +1,161 @@
+		time.Sleep(pauseDuration)
+
+		if err = vt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		asset, err := vt.lapi.GetAsset(importAsset.ID)
+		if err != nil {
+			glog.Errorf("Error retrieving asset id=%s err=%v", importAsset.ID, err)
+			return 243, err
+		}
+		if asset.Status.Phase == "ready" {
+			break
+		}
+		if asset.Status.Phase != "waiting" {
+			glog.Errorf("Error importing asset assetId=%s, task id=%s err=%v", importAsset.ID, importTask.ID, err)
+			return 244, err
+		}
+	}
+
+	transcodeAsset, transcodeTask, err := vt.lapi.TranscodeAsset(importAsset.ID, assetName, api.StandardProfiles[0])
+	if err != nil {
+		glog.Errorf("Error creating transcode task err=%v", err)
+		return 242, err
+	}
+	glog.Infof("Asset imported id=%s, transcoding taskId=%s outputAssetId=%s", importAsset.ID, transcodeTask.ID, transcodeAsset.ID)
+
+	startTime = time.Now()
+	for {
+		glog.Infof("Waiting %s for transcode assetId=%s outputAssetId=%s, elapsed=%s", pauseDuration, importAsset.ID, transcodeAsset.ID, time.Since(startTime))
+		time.Sleep(pauseDuration)
+
+		if err = vt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		asset, err := vt.lapi.GetAsset(transcodeAsset.ID)
+		if err != nil {
+			glog.Errorf("Error retrieving asset id=%s err=%v", transcodeAsset.ID, err)
+			return 243, err
+		}
+		if asset.Status.Phase == "ready" {
+			break
+		}
+		if asset.Status.Phase != "waiting" {
+			glog.Errorf("Error transcoding asset assetId=%s, task id=%s outputAssetId=%s err=%v", importAsset.ID, transcodeTask.ID, transcodeAsset.ID, err)
+			return 244, err
+		}
+	}
+
+	exportTask, err := vt.lapi.ExportAsset(importAsset.ID)
+	if err != nil {
+		glog.Errorf("Error creating export task err=%v", err)
+		return 245, err
+	}
+	glog.Infof("Transcode complete assetId=%s ready, exporting assetId=%s, taskId=%s", transcodeAsset.ID, importAsset.ID, exportTask.ID)
+
+	startTime = time.Now()
+	for {
+		glog.Infof("Waiting %s for asset id=%s to be exported, elapsed=%s", pauseDuration, transcodeAsset.ID, time.Since(startTime))
+		time.Sleep(pauseDuration)
+		if err = vt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		task, err := vt.lapi.GetTask(exportTask.ID)
+		if err != nil {
+			glog.Errorf("Error retrieving task id=%s err=%v", exportTask.ID, err)
+			return 246, err
+		}
+		if task.Output.Export.IPFS.VideoFileGatewayUrl != "" {
+			glog.Infof("Export success, task id=%s assetId=%s ipfs link=%s", exportTask.ID, exportTask.InputAssetID, task.Output.Export.IPFS.VideoFileGatewayUrl)
+			break
+		}
+	}
+
+	glog.Info("Done VOD Test")
+
+	return 0, err
+}
+
+func (vt *vodTester) isCancelled() error {
+	select {
+	case <-vt.ctx.Done():
+		return context.Canceled
+	default:
+	}
+	return nil
+}
+
+func (vt *vodTester) Cancel() {
+	vt.cancel()
+}
+
+func (vt *vodTester) Done() <-chan struct{} {
+	return vt.ctx.Done()
+}
+
+func (rt *vodTester) Clean() {
+	// TODO
+}
```

Comment:
Instead of having this clean function, you could simply do a `defer api.DeleteAsset(asset.ID)` after every asset you created on the test routine above.

We probably don't want to infinitely stack up the account with so many assets from these tests. Although we won't be really deleting the files from cloud storage nor even the objects from DB anyway 🤷
So it's more not to pollute the API responses/dashboard than for any concrete resource optimization.